javascript/javascript1/week2/candy-helper.js

```diff
@@ -0,0 +1,49 @@
+const boughtCandyPrices = [];
+var amountToSpend = Math.random() * 100;
+
+function addCandy(candyType, weight) {
+    switch (candyType) {
+        case 'Sweet':
+            return boughtCandyPrices.push(candyType, weight * 0.5);
+            break;
+        case 'Chocolate':
+            return boughtCandyPrices.push(candyType, weight * 0.7);
+            break;
+        case 'Toffee':
+            return boughtCandyPrices.push(candyType, weight * 1.1);
+            break;
+        case 'Chewing-gum':
+            return boughtCandyPrices.push(candyType, weight * 0.03);
+            break;
+        default:
+            return console.log("Item not available");
+            break;
+    }
+}
+
+function canBuyMoreCandy() {
+    let canBuyMore;
+    let prices = boughtCandyPrices.filter((element, index) => {
+            return index % 2 !== 0;
+        }) //All odd indexed values stored into prices a array
+    var total = 0;
+    for (var i in prices) {
+        total += prices[i];
+    } // for loop can be replaced by: prices.reduce((x, y) => x + y);
```

Comment:
You can simply calculate the total using for loop or `for .. of`
```js
  let total = 0;
  for (let i = 0; i < boughtCandyPrices.length; i++) {
    total += boughtCandyPrices[i];
  }
```

Reply:
let total = 0;
    for (let i = 0; i < boughtCandyPrices.length; i++) {
        if (i % 2 !== 0) {
            total += boughtCandyPrices[i];
        }
    }